include/eve/module/core/constant/eps.hpp

```diff
@@ -7,22 +7,37 @@
 //==================================================================================================
 #pragma once
 
-#include <eve/as.hpp>
-#include <eve/concept/value.hpp>
-#include <eve/detail/implementation.hpp>
-#include <eve/detail/meta.hpp>
-#include <eve/module/core/constant/constant.hpp>
-#include <eve/module/core/decorator/roundings.hpp>
-
-#include <type_traits>
+#include <eve/arch.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
 
 namespace eve
 {
+template<typename Options>
+struct eps_t : constant_callable<eps_t, Options, downward_option, upward_option>
+{
+  template<typename T>
+  static EVE_FORCEINLINE constexpr T value(eve::as<T> const&, auto const&)
+  {
+    using e_t = element_type_t<T>;
+
+    if      constexpr(std::integral<e_t>        ) return T(1);
```

Comment:
eps 1 for integral looks a bit weird? Can we maybe just have `eps` for floating point? Or is this how numeric_limits define ti?

On that note - should we forward to std::numeric_limits?

Reply:
This is the generalized epsilon defined as the smallest positive value so that `1 + eps != 1` and this definition works for integers